.github/workflows/linux-clang8-debug.yml

```diff
@@ -0,0 +1,300 @@
+name: "Linux build, clang-8 debug unity"
+on: [push, pull_request]
+
+# The files starting with "linux-" are all interdependent.
+# If any significant changes are made, be sure to update them
+# to keep them in sync.
+
+env:
+  CMAKE_EXTRA_ARGS: "-Dwerr=ON -Dwextra=ON"
+  NINJA_BUILD: true
+  CACHE_DIR: ${{ github.workspace }}/_cache
+  NIH_CACHE_ROOT: ${{ github.workspace }}/_cache/nih_c
+  PARALLEL_TESTS: true
+  USE_CCACHE: true
+  CCACHE_BASEDIR: ${{ github.workspace }}
+  CCACHE_NOHASHDIR: true
+  CCACHE_DIR: ${{ github.workspace }}/_cache/ccache
+
+jobs:
+  linux_clang_8_debug_unity:
+    name: Linux
+    runs-on: ubuntu-18.04
+    timeout-minutes: 720
+    if: ${{ !contains(github.event.head_commit.message, 'ci_run_') || contains(github.event.head_commit.message, 'ci_run_linux') }}
+    outputs:
+      cmakeconfig: ${{ steps.cachekeys.outputs.cmakeconfig }}
+      headers: ${{ steps.cachekeys.outputs.headers }}
+      source: ${{ steps.cachekeys.outputs.source }}
+      instructions: ${{ steps.cachekeys.outputs.instructions }}
+    strategy:
+      fail-fast: false
+      matrix:
+        CC: [ "clang-8" ]
+        BUILD_TYPE: [ Debug ]
+        unity: [ "ON" ]
+        image: [ "rippled-ci-builder:2020-01-08" ]
+        name: [ "base" ]
+    container:
+      image: rippleci/${{ matrix.image }}
+    env:
+      CC: ${{ matrix.CC }}
+      BUILD_TYPE: ${{ matrix.BUILD_TYPE }}
+      CMAKE_ADD: "-Dunity=${{ matrix.unity }} ${{ matrix.CMAKE_ADD }}"
```

Comment:
That looks like a copy/paste artifact from linux-general.yml. It doesn't hurt to have it there - unset matrix variables just act like an empty string. I'm sure there are more than a few of these floating around because while working on it, I wasn't always sure where I'd need to add things like `CMAKE_ADD` back later.

Reply:
There's now only one value for each variable in the matrix, meaning a matrix is not necessary. I'm starting to wonder, though, if we can combine all these `linux-$compiler-$flavor.yml` workflows into one with a combined matrix. Before I dive in, what should I know? (This kind of architectural documentation should probably go in a README too.)

Reply:
I initially created the "base" linux builds as a single large matrix in one file. It worked fine, with the exception that if any job failed spuriously, the entire workflow had to be restarted. Obviously, that shouldn't happen often, but when it does, it's pretty annoying. Breaking them up into separate files / workflows helped increase the granularity. Another downside of having them in one matrix is that the dependent jobs (manual tests, etc.) had to wait for _all_ of the "base" jobs to finish. Splitting them into separate jobs isn't much different than separating them into separate files, other than the restart granularity.

I'm pretty confident that, _all_ the jobs (linux, mac, & win) could be combined into a single workflow. I guess I didn't document that, but I figured that if Github ever adds the ability to restart a single job, it would be worth combining them.

Reply:
Oh, and the reason I used a matrix with one value each is that I ran into problems setting those values as environmental variables and using them in non-shell steps. I'll look at it again to check if that's still the case.